packages/dynamite/dynamite_runtime/lib/src/client/client.dart

```diff
@@ -130,6 +122,17 @@ class DynamiteClient with http.BaseClient {
       await cookieJar!.saveFromResponse(request.url, cookies);
     }
 
+    if (request case DynamiteRequest(:final validStatuses)
+        when validStatuses != null && !validStatuses.contains(response.statusCode)) {
```

Comment:
First I've never seen this syntax :sweat_smile: and it also is not very readable to have it all in the if statement

Reply:
Dart3 pattern matching :P

Do you really think the usual if is easier to read?
```dart
    if (request is DynamiteRequest) {
      final validStatuses = request.validStatuses;
      if (validStatuses != null && !validStatuses.contains(response.statusCode)) {
        
      }
    }
```
or as we do not need the validStatuses itself
```dart
    if (request is DynamiteRequest && request.validStatuses != null 
      && !request.validStatuses!.contains(response.statusCode)) {
```

`validStatuses` is a getter so we do not get null check elevation.
I agree that it might not be strictly necessary here but I would also not be against it in general. It can help reduce a lot of boilerplate code.

Reply:
Either of those options are better than the current one imo.